vector/src/main/java/im/vector/app/features/spaces/SpaceSettingsMenuBottomSheet.kt

```diff
@@ -138,8 +148,38 @@ class SpaceSettingsMenuBottomSheet : VectorBaseBottomSheetDialogFragment<BottomS
         }
 
         views.leaveSpace.views.bottomSheetActionClickableZone.debouncedClicks {
+            val spaceSummary = activeSessionHolder.getSafeActiveSession()?.getRoomSummary(spaceArgs.spaceId)
+                    ?: return@debouncedClicks
+            val warningMessage: CharSequence = if (spaceSummary.otherMemberIds.isEmpty()) {
+                span {
+                    +getString(R.string.space_leave_prompt_msg)
+                    +"\n"
+                    span(getString(R.string.space_leave_prompt_msg_only_you)) {
+                        textColor = ContextCompat.getColor(requireContext(), R.color.riotx_destructive_accent)
+                    }
+                }
+            } else if (isAdmin) {
+                span {
+                    +getString(R.string.space_leave_prompt_msg)
+                    +"\n"
+                    span(getString(R.string.space_leave_prompt_msg_as_admin)) {
+                        textColor = ContextCompat.getColor(requireContext(), R.color.riotx_destructive_accent)
+                    }
+                }
+            } else if (!spaceSummary.isPublic) {
+                span {
+                    +getString(R.string.space_leave_prompt_msg)
+                    +"\n"
+                    span(getString(R.string.space_leave_prompt_msg_private)) {
+                        textColor = ContextCompat.getColor(requireContext(), R.color.riotx_destructive_accent)
+                    }
+                }
+            } else {
+                getString(R.string.space_leave_prompt_msg)
+            }
+
             AlertDialog.Builder(requireContext())
-                    .setMessage(getString(R.string.space_leave_prompt_msg))
+                    .setMessage(warningMessage)
```

Comment:
When we leave a room, the positive button is colored in red in case of warning. Do we want to do the same for space?

Reply:
done, plus refactored a bit to use title and message instead of adding \n